src/main/java/org/eclipsefoundation/marketplace/model/Market.java

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright (C) 2019 Eclipse Foundation and others.
+ * 
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ * 
+ * SPDX-License-Identifier: EPL-2.0
+*/
+package org.eclipsefoundation.marketplace.model;
+
+import java.security.InvalidParameterException;
+import java.util.ArrayList;
+import java.util.LinkedList;
+import java.util.List;
+
+import io.quarkus.runtime.annotations.RegisterForReflection;
+
+/**
+ * Domain object representing a Market within the marketplace.
+ * 
+ * @author Martin Lowe
+ * @since 05/2019
+ */
+@RegisterForReflection
+public class Market {
+	private int id;
+	private String name;
+	private String url;
+	private List<Category> categories;
+
+	/**
+	 * Default constructor. Creates an empty linkedlist for categories, as its
+	 * unknown how many categories the market will reference.
+	 */
+	public Market() {
+		this.categories = new LinkedList<>();
```

Comment:
Why do you us a `LinkedList`? Do you plan to insert items at given id? If no, then an `ArrayList` if more than enough.
I get the `removeCategory(Category)` use case, but I think you would be better served by

    public boolean removeCategory(Category category) {
               return categories.removeIf(c -> c.equals(category));
     }

Reply:
In previous projects, I used LinkedList when I didn't know exact size of an array. While the upfront cost is a little higher, its much more efficient at maintaining large scale lists due to no growth operations being required. Whats the rule that you generally follow around this?